### PR TITLE
[Backend] Add cron job to stop workers for unapproved challenges

### PR DESCRIPTION
**Pull request by Ram81**
### Description

This PR adds -

- [x] Changes to cron job to delete workers for challenges that have been unapproved for 3 days
- [x] API changes to delete workers only by EvalAI admins
- [x] Bug fix in existing logic to delete workers after challenge ends

### CODE REVIEW

File: scripts/monitoring/auto_start_or_stop_workers.py [hunk 1]
```diff
@@ -46,32 +58,54 @@ def get_all_challenges():
...
+        approved_by_admin = challenge["approved_by_admin"]
         is_docker_based = challenge["is_docker_based"]
         challenge_start_date = parser.parse(challenge["start_date"])
         challenge_end_date = parser.parse(challenge["end_date"])
+        created_at = parser.parse(challenge["created_at"])
         current_date = utc.localize(datetime.now())
 
         print(
-            "Start submission worker for challenge id {}".format(challenge_id)
+            "Start/Stop submission worker for challenge id {}".format(
```
Comment (RishabhJain2018) [inline scripts/monitoring/auto_start_or_stop_workers.py:82]: Can we please be specific with the message? Also, why are we using `print` statement? Would this be logged on Cloudwatch?

File: scripts/monitoring/auto_start_or_stop_workers.py [hunk 2]
```diff
@@ -46,32 +58,54 @@ def get_all_challenges():
...
                 response = start_worker(challenge_id)
                 if not response.ok:
                     print(
                         "ERROR: Start worker failed for challenge id {}!".format(
                             challenge_id
                         )
                     )
-            if workers is not None and challenge_end_date <= current_date:
-                response = stop_worker(challenge_id)
+            # Delete workers for challenges uploaded in last 3 days that are unapproved or inactive challenges
```
Comment (RishabhJain2018) [inline scripts/monitoring/auto_start_or_stop_workers.py:99]: Please make it 24 hours.

File: scripts/monitoring/auto_start_or_stop_workers.py [hunk 3]
```diff
@@ -46,32 +58,54 @@ def get_all_challenges():
...
+            if (
+                workers is not None and challenge_end_date < current_date
+            ) or is_unapproved_challenge(
+                workers, approved_by_admin, created_at
+            ):
+                response = delete_worker(challenge_id)
                 if not response.ok:
                     print(
-                        "ERROR: Stop worker failed for challenge id {}!".format(
+                        "ERROR: Delete worker failed for challenge id {}!".format(
```
Comment (RishabhJain2018) [inline scripts/monitoring/auto_start_or_stop_workers.py:108]: We should trigger sentry. I don't think `print` will help us here as we won't check the logs manually.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

File: scripts/monitoring/auto_start_or_stop_workers.py (hunk 1 above)
Comment (Ram81) [reply scripts/monitoring/auto_start_or_stop_workers.py:82]: This is a independent cron that we run on prod and staging instance. Do we want the logs on cloudwatch?

**Comment by Ram81** [comment]
@RishabhJain2018 this is a independent cron that I was planning to run in `crontab` on the instance. Do we want this to run inside django container?

**Comment by RishabhJain2018** [comment]
Ummm..then how do we catch errors from this cron job?

**Comment by Ram81** [comment]
The logs are available in crontab. If something is wrong then I just check there. As these are fixed API calls which are already validated beforehand there have been no errors so far.

**Comment by RishabhJain2018** [comment]
okay.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**